src/cli/rustup_mode.rs

```diff
@@ -939,7 +940,8 @@ fn update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
                 let host_arch = TargetTriple::from_host_or_build();
                 if let Ok(toolchain_desc) = ToolchainDesc::from_str(name) {
                     let target_triple = toolchain_desc.target;
-                    if host_arch.ne(&target_triple) {
+                    if !forced && host_arch.ne(&target_triple) {
+                        err!("DEPRECATED: please use --force to permit this in future releases of rustup.");
```

Comment:
Maybe make it clear that the behavior is allowed currently? But the current text seems fine too.
```suggestion
                        err!("DEPRECATED: future versions of rustup will require --force to install a non-host toolchain as the default");
```